supporter-product-data/cloudformation/cfn.yaml

```diff
@@ -0,0 +1,383 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: "Product data information for supporters."
+
+Parameters:
+  Stage:
+    Description: Stage name
+    Type: String
+    AllowedValues:
+      - PROD
+      - UAT
+      - CODE
+    Default: CODE
+
+Conditions:
+  CreateProdResources: !Equals [!Ref "Stage", "PROD"]
+
+Mappings:
+  StageVariables:
+    CODE:
+      S3Bucket: supporter-product-data-export-code
+      # Every 10 mins between 7am and 7pm Mon-Fri
+      scheduleRate: "cron(*/10 7-19 ? * 1-5 *)"
+    UAT:
+      S3Bucket: supporter-product-data-export-uat
+      scheduleRate: "cron(*/10 7-19 ? * 1-5 *)"
+    PROD:
+      S3Bucket: supporter-product-data-export-prod
+      # Every 5 mins
+      scheduleRate: "cron(*/5 * ? * * *)"
+
+Resources:
+
+  SupporterProductExportBucket:
+    Type: "AWS::S3::Bucket"
+    Properties:
+      BucketName:
+        !FindInMap [StageVariables, !Ref Stage, S3Bucket]
+      AccessControl: Private
+      LifecycleConfiguration:
+        Rules:
+          - Id: DeleteAllOldFiles
+            Prefix: ""
+            Status: Enabled
+            ExpirationInDays: 365
+          - Id: DeleteOldFilesZuora
+            Prefix: "zuora"
+            Status: Enabled
+            ExpirationInDays: 14
+      VersioningConfiguration:
+        Status: "Enabled"
+
+  EncryptBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    DependsOn: SupporterProductExportBucket
+    Properties:
+      Bucket: !Sub ${SupporterProductExportBucket}
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: DenyIncorrectEncryptionHeader
+            Effect: Deny
+            Principal: "*"
+            Action: s3:PutObject
+            Resource: !Sub arn:aws:s3:::${SupporterProductExportBucket}/*
+            Condition:
+              StringNotEquals:
+                s3:x-amz-server-side-encryption:
+                  - AES256
+                  - aws:kms
```

Comment:
why kms rather than s3 managed encryption?
I think s3 managed encryption is cheaper from a quick look

Reply:
Again, copied from fulfilment-lambdas, I've changed it now.